src/smt.cpp

```diff
@@ -20,25 +21,40 @@
 
 using namespace std;
 
-namespace {
-z3::expr_vector toZ3ExprVector(const vector<z3::expr> &vec) {
-  z3::expr_vector ev(*smt::sctx.z3);
-  for (auto &e: vec)
-    ev.push_back(e);
-  return ev;
-}
 
-z3::expr_vector toZ3ExprVector(const vector<smt::Expr> &vec) {
-  z3::expr_vector ev(*smt::sctx.z3);
-  for (auto &e: vec)
-    ev.push_back(e.getZ3Expr());
-  return ev;
-}
+namespace {
+z3::expr_vector toZ3ExprVector(const vector<z3::expr> &vec);
+z3::expr_vector toZ3ExprVector(const vector<smt::Expr> &vec);
 }
 
 namespace smt {
+class Context {
+private:
+  unordered_map<string, uint64_t> fresh_var_map;
+
+public:
+  IF_Z3_ENABLED(optional<z3::context> z3);
+  IF_CVC5_ENABLED(optional<cvc5::api::Solver> cvc5);
+
+  uint64_t timeout_ms;
+
+  Context() {
+    IF_Z3_ENABLED(this->z3.reset());
+    IF_CVC5_ENABLED(this->cvc5.reset());
+    timeout_ms = 20000;
+  }
+
+  IF_Z3_ENABLED(void useZ3() { this->z3.emplace(); })
+  IF_CVC5_ENABLED(void useCVC5() { this->cvc5.emplace(); })
 
-Context sctx(true);
+  string getFreshName(string prefix) {
```

Comment:
Is this related to Context?

Reply:
CVC5 does not natively support fresh naming, so use our own semantics